kit/src/main/java/com/oracle/javafx/scenebuilder/kit/skeleton/SkeletonFileWriter.java

```diff
@@ -0,0 +1,228 @@
+    }
+
+    /**
+     * Prepares a language specific file name proposal for a controller skeleton
+     * file. Then, a file chooser is shown to allow adjustments to the proposed file
+     * name. Finally, the contents of the provided textProperty is saved to the
+     * file. In case of success, a message is shown - same applies for the error
+     * case.
+     * 
+     * The file chooser used will remember the name of the saved file. To create a
+     * new file name proposal, the FXML file must be closed and re-opened in
+     * SceneBuilder.
+     * 
+     */
+    public void run() {
+
+        if (saveFileChooser == null) {
+            saveFileChooser = new FileChooser();
+        }
+
+        if (savedFile == null) {
+            File fileToSave = new SkeletonFileNameProposal(language).create(fxmlLocation, controllerName);
+            updateFileChooser(fileToSave);
+        }
+        
+        /*
+         * TODO:
+         * 
+         * Clarify when to update the Skeleton file name proposal. Currently, the last
+         * saved file name is remembered. The file name will only be updated after
+         * re-opening the FXML file. The file name and its location are store in the
+         * file chooser instance.
+         * 
+         */
+
+        Platform.runLater(() -> {
+            File confirmedSavedFile = saveFileChooser.showSaveDialog(stageSupplier.get());
+            if (null != confirmedSavedFile) {
+                savedFile = confirmedSavedFile;
+                updateFileChooserAndSave(confirmedSavedFile);
+            }
+        });
+    }
+
+    private void updateFileChooserAndSave(File savedFile) {
+        updateFileChooser(savedFile);
+        writeSkeletonToFile(savedFile.toPath().toAbsolutePath());
+    }
+
+    private void updateFileChooser(File fileToSave) {
+        saveFileChooser.setInitialDirectory(fileToSave.getParentFile());
+        saveFileChooser.setInitialFileName(fileToSave.getName());
+    }
+
+    private void writeSkeletonToFile(Path skeletonFile) {
+        try {
+            attemptWritingSkeletonFile(skeletonFile);
+        } catch (IOException e) {
+            showAlertOnErrorWritingSkeletonFile(skeletonFile, e);
+        }
+    }
+
+    private void attemptWritingSkeletonFile(Path skeletonFile) throws IOException {
+
+        String skeleton = textProperty.getValueSafe();
+        Files.writeString(skeletonFile, skeleton, StandardOpenOption.CREATE);
+
+        var alert = new SBAlert(AlertType.INFORMATION, stageSupplier.get());
+        alert.setContentText("The controller skeleton file was successfully written.");
+        
+        DialogPane dialogPane = alert.getDialogPane();       
+        TextField textField  = new TextField(skeletonFile.toString());
+        Button    copyButton = new Button(I18N.getString("label.copy")); 
+        
+        HBox hbox = new HBox(textField,copyButton);
+        textField.setMaxWidth(600);
+        textField.setPrefWidth(300);
```

Comment:
To avoid resizing of the Alert, I would remove setting width on the TextField.

Reply:
This is removed.

Reply:
Well, I am not sure on that. Resizeability is not necessarily bad. Especially with long path names, sometimes a wider dialog can be nice. It's not resizeable at this moment. One option is to provide some extended default for the alert, 500px or 600px.